### PR TITLE
Switch s3 sidecar pipeline detail fetch back to gRPC

### PR DESCRIPTION
**Pull request by PFedak**
Also run s3 tests with auth enabled by default.

### CODE REVIEW

File: etc/testing/circle/run_tests.sh [hunk 1]
```diff
@@ -115,7 +116,7 @@ case "${BUCKET}" in
     make test-enterprise
     # Launch a stand-alone enterprise server in a separate namespace
     make launch-enterprise
-    echo "{\"pachd_address\": \"grpc://${VM_IP}:${ENTERPRISE_PORT}\", \"source\": 2}" | pachctl config set context "enterprise" --overwrite
```
Comment (acohen4) [inline etc/testing/circle/run_tests.sh:118]: this is just formatting right?

**Review by acohen4 — APPROVED** [review]
Change looks good! I just don't understand why S3G needs to communicate over gRPC. Could you explain?

**Comment by PFedak** [comment]
It doesn't actually need to, I should be a bit clearer. It's just a silly thing with context metadata. We put the auth token in the outgoing metadata for the grpc to read and place in the new context's incoming metadata. Our checks look at the incoming metadata, so if you remove the gRPC from the process they don't find auth info. It could also be fixed by copying the metadata over.

File: etc/testing/circle/run_tests.sh (hunk 1 above)
Comment (PFedak) [reply etc/testing/circle/run_tests.sh:118]: yes, I guess my editor did that automatically.

**Pull request merged**